warehouse/integrations/middleware/sqlquerywrapper/sql.go

```diff
@@ -108,6 +109,28 @@ func (db *DB) QueryRowContext(ctx context.Context, query string, args ...interfa
 	return row
 }
 
+func (db *DB) WithTx(fn func(*Tx) error, timeout time.Duration) error {
```

Comment:
What about having a context propagated here as well? I haven't checked how extensive the change might be but it may be worth considering. If not in this PR perhaps it can be addressed in a dedicated PR. Wdyt?